src/components/back-link-helper/back-link-helper.js

```diff
@@ -14,10 +14,15 @@ BackLinkHelper.prototype.init = function init() {
        Currently, a page refresh sets the referer to empty, leading to the back link being hidden
        under our existing logic.
      */
-    // eslint-disable-next-line max-len
     const referrerNotOnSameDomain = () => {
       const referer = this.document.referrer;
-      return !referer || referer.indexOf(this.window.location.host) === -1;
+      const noReferer = !referer || referer.indexOf(this.window.location.host) === -1;
+
+      // Allows PEGA domains
+      const allowedDomains = ['maccount-np.hmrc.gov.uk', 'account.hmrc.gov.uk', 'https://catalogue.tax.service.gov.uk/'];
+      const isAllowed = (!noReferer) ? allowedDomains.some((e) => referer.includes(e)) : false;
+
+      return (isAllowed) ? false : noReferer;
     };
```

Comment:
the logic of this is quite hard to read with the multiple conditionals, noreferer now also contains logic to check "no referer or referer is a different domain", what about something like:

```javascript
    const referer = this.document.referrer;

    const referrerNotOnSameDomain = () =>
      (referer.indexOf(this.window.location.host) === -1);
    
    const allowedDomains = ...
    const referrerNotOnAllowList = () =>
      !allowedDomains.some(allowedDomain => referer.includes(allowedDomain));

    if (!referrer || (referrerNotOnSameDomain() && referrerNotOnAllowList())) {
      this.$module.classList.add('hmrc-hidden-backlink');
```

might have some syntactic mistakes, just written out in comment - would still need some automated tests it works